dotnet-interactive.Tests/InMemoryJupyterKernelSpecInstaller.cs

```diff
@@ -4,55 +4,55 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Threading.Tasks;
-using Microsoft.DotNet.Interactive.Utility;
 
 namespace Microsoft.DotNet.Interactive.App.Tests
 {
-    internal class InMemoryJupyterKernelSpec : IJupyterKernelSpec
+    internal class InMemoryJupyterKernelSpecInstaller : IJupyterKernelSpecInstaller
     {
         private readonly bool _shouldInstallSucceed;
         private readonly bool _shouldUninstallSucceed;
-        private readonly IReadOnlyCollection<string> _error;
+        private readonly string _message;
 
-        public InMemoryJupyterKernelSpec(
+        public InMemoryJupyterKernelSpecInstaller(
             bool shouldInstallSucceed,
-            IReadOnlyCollection<string> error,
+            string message,
             bool shouldUninstallSucceed = true)
         {
             _shouldInstallSucceed = shouldInstallSucceed;
             _shouldUninstallSucceed = shouldUninstallSucceed;
-            _error = error;
+            _message = message;
         }
 
         public List<string> InstalledKernelSpecs { get; } = new List<string>();
 
-        public Task<CommandLineResult> InstallKernel(DirectoryInfo directory)
+        public Task<KernelSpecInstallResult> InstallKernel(DirectoryInfo kernelSpecPath, DirectoryInfo destination = null)
```

Comment:
Does this method need to return `Task`?

Reply:
this is used in the command to print out to console

Reply:
I mean if all return paths need do `Task.FromRestult` then it looks like there's no need to wrap the `KernelSpecInstallResult` in a `Task`.